ArduPlane/system.cpp

```diff
@@ -526,6 +526,60 @@ void Plane::exit_mode(enum FlightMode mode)
         auto_state.started_flying_in_auto_ms = 0;
     }
 }
+// check_swarm_failsafe - This enables failsafe on loss of both RC input and GCS link. If THR_FAILSAFE is 0,this will only detect GCS failsafe.It only works in Guided Mode
+void Plane::check_swarm_failsafe()
+{
+	uint32_t tnow = millis();
+	int16_t failsafe_counts = 0;
+	// exit if not in guided mode or rtl mode
+	// TODO:We can consider the sitution in auto mode or other position control mode
+	if (!g.swarm_failsafe_enable || (control_mode != GUIDED && control_mode != RTL)) {
+		return;
+	}
+	if (g.throttle_fs_enabled) {
```

Comment:
lots of redundant checking here just to make it ++ vs +=2. How about 
```
if (g.throttle_fs_enabled) {
    failsafe_count_inc = 1;
else {
    failsafe_count_inc = 2;
}

```
and then have the other code say
`failsafe_counts += failsafe_counts_inc
`